### PR TITLE
Remove textarea caret dep

### PR DESCRIPTION
**Pull request by witmicko**
## Explanation

At the screen where the user inserts their password, every keystroke that occurs triggers textarea-caret dependency
If textarea-caret is breached it can be bad, a third party package should not be triggered in such a sensitive phase.
This fix replaces this dependency with our own less generic implementation
Fixes Logo head following password as it is being typed
Fixes  [cirt:#114](https://github.com/MetaMask/pm-security/issues/114)

## Manual Testing Steps

lock metamask, start typing in password, verify that the logo head is following latest character correctly.
verify different browsers and resolutions

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/unlock-page/unlock-page.util.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  div.textContent = new Array(position + 1).join('•');
+  const span = document.createElement('span');
+  span.textContent = '•';
+  div.appendChild(span);
+
+  const coordinates = {
+    top: span.offsetTop + parseInt(computed.borderTopWidth, 10),
+    left: span.offsetLeft + parseInt(computed.borderLeftWidth, 10),
+  };
+  document.body.removeChild(div);
```
Comment (weizman) [inline ui/pages/unlock-page/unlock-page.util.js:15]: i would do `div.remove()` instead so the removal operation will be independent of who's the parent

**Review by weizman — DISMISSED** [review]
LGTM

File: ui/pages/unlock-page/unlock-page.util.js (hunk 1 above)
Comment (kingdutch001) [reply ui/pages/unlock-page/unlock-page.util.js]: DIV

Comment (kingdutch001) [reply ui/pages/unlock-page/unlock-page.util.js]: i would do div.remove() instead so the removal operation will be independent of who's the parent

**Review by weizman — APPROVED** [review]
lgtm

**Review by pedronfigueiredo — APPROVED** [review]

**Review by danjm — COMMENTED** [review]
Nice work

**Pull request merged**